src/test/java/brave/opentracing/BraveTracerTest.java

```diff
@@ -188,4 +196,80 @@ void checkSpanReportedToZipkin() {
         }
     );
   }
+
+  @Test public void subsequentChildrenNestProperly_OTStyle() {
+    // this test is semantically identical to subsequentChildrenNestProperly_BraveStyle, but uses
+    // the OpenTracingAPI instead of the Brave API.
+
+    Long idOfSpanA;
+    Long shouldBeIdOfSpanA;
+    Long idOfSpanB;
+    Long shouldBeIdOfSpanB;
+    Long parentIdOfSpanB;
+    Long parentIdOfSpanC;
+
+    try (ActiveSpan spanA = opentracing.buildSpan("spanA").startActive()) {
+      idOfSpanA = getTraceContext(spanA).spanId();
+      try (ActiveSpan spanB = opentracing.buildSpan("spanB").startActive()) {
+        idOfSpanB = getTraceContext(spanB).spanId();
+        parentIdOfSpanB = getTraceContext(spanB).parentId();
+        shouldBeIdOfSpanB = getTraceContext(opentracing.activeSpan()).spanId();
+      }
+      shouldBeIdOfSpanA = getTraceContext(opentracing.activeSpan()).spanId();
+      try (ActiveSpan spanC = opentracing.buildSpan("spanC").startActive()) {
+        parentIdOfSpanC = getTraceContext(spanC).parentId();
+      }
+    }
+
+    assertEquals("SpanA should have been active again after closing B", idOfSpanA, shouldBeIdOfSpanA);
+    assertEquals("SpanB should have been active prior to its closure", idOfSpanB, shouldBeIdOfSpanB);
+    assertEquals("SpanB's parent should be SpanA", idOfSpanA, parentIdOfSpanB);
+    assertEquals("SpanC's parent should be SpanA", idOfSpanA, parentIdOfSpanC);
+  }
+
+  @Test public void subsequentChildrenNestProperly_BraveStyle() {
```

Comment:
a mixed test would be nice to show how they work together. Also unit tests especially for the racy code in package private classes so they don't bite later.